tethys_apps/models.py

```diff
@@ -229,9 +235,153 @@ def get_value(self, *args, **kwargs):
+
+    ::
+
+        from tethys_sdk.app_settings import SecretCustomSetting
+
+        default_name_setting = SecretCustomSetting(
+            name='Github_Token',
+            description='Github repository secret token',
+            required=True,
+        )
+
+        max_count_setting = SecretCustomSetting(
+            name='Anaconda_Token',
+            description='Anaconda Organization secret token',
+            required=False
+        )
+    """  # noqa: E501
+
+    value = models.CharField(max_length=1024, blank=True, default="")
+
+    def clean(self):
+        """
+        Validate prior to saving changes.
+        """
+
+        if type(self.value) is not str:
+            raise ValidationError(
+                "Validation Error: Secret Custom Setting should be a String"
+            )
+        else:
+            try:
+                json.loads(self.value)
+                raise ValidationError(
+                    "Validation Error: Secret Custom Setting should not be a JSON String"
+                )
+            except ValueError:
+                pass
+
+        if self.value == "" and self.required:
+            raise ValidationError("Required.")
+        if self.value != "":
+            TETHYS_HOME = get_tethys_home_dir()
+            signer = Signer()
+            if not os.path.exists(os.path.join(TETHYS_HOME, "secrets.yml")):
+                self.value = signer.sign_object(self.value)
+            else:
+                with open(os.path.join(TETHYS_HOME, "secrets.yml")) as secrets_yaml:
+                    secret_app_settings = (
+                        yaml.safe_load(secrets_yaml).get("secrets", {}) or {}
+                    )
+                    if bool(secret_app_settings):
+                        if self.tethys_app.package in secret_app_settings:
+                            if (
+                                "custom_settings_salt_strings"
+                                in secret_app_settings[self.tethys_app.package]
+                            ):
+                                app_specific_settings = secret_app_settings[
+                                    self.tethys_app.package
+                                ]["custom_settings_salt_strings"]
+                                if self.name in app_specific_settings:
+                                    app_custom_setting_salt_string = (
+                                        app_specific_settings[self.name]
+                                    )
+                                    if app_custom_setting_salt_string != "":
+                                        signer = Signer(
+                                            salt=app_custom_setting_salt_string
+                                        )
+
+                        self.value = signer.sign_object(self.value)
+
+                    else:
+                        log.info(
+                            "There is not a an apps portion in the secrets.yml, please create one."
```

Comment:
Should this read: "There is not a **secrets** section in the secrets.yml..."?

Reply:
Done